lib/appium.js

```diff
@@ -178,24 +183,30 @@ class AppiumDriver extends BaseDriver {
       }
     }
 
-    let runningDriversData;
-    try {
-      runningDriversData = this.curSessionDataForDriver(InnerDriver);
-    } catch (e) {
-      throw new errors.SessionNotCreatedError(e.message);
-    }
-
-    let innerSessionId, dCaps;
+    let runningDriversData, otherPendingDriversData;
     let d = new InnerDriver(this.args);
-    try {
+    await sessionCreationLock.acquire(AppiumDriver.name, async () => {
```

Comment:
I see it as the first step, that prepares AppiumDriver to by truly async and allow to execute multiple commands in parallel, without scheduling.

For now these changes don't make any difference, because each createSession (and other belonging to AppiumDriver) command is put into commands queue and I'd like to change this behaviour.